android/app/src/main/java/com/zulipmobile/sharing/ReceiveShareActivity.kt

```diff
@@ -0,0 +1,129 @@
+        params.putString("sharedText", sharedText)
+        if (sharedText != null) {
+            if (reactContext == null) {
+                SharingModule.initialSharedData = params
+            } else {
+                sendEvent(reactContext, "shareReceived", params)
+            }
+
+        }
+    }
+
+    private fun handleSendImage(reactContext: ReactContext?, intent: Intent) {
+        (intent.getParcelableExtra<Parcelable>(Intent.EXTRA_STREAM) as? Uri)?.let {
+            val params = Arguments.createMap()
+            params.putString("type", "image")
+            params.putString("sharedImageUri", it.toString())
+            if (reactContext == null) {
+                SharingModule.initialSharedData = params
+            } else {
+                sendEvent(reactContext, "shareReceived", params)
+            }
+        }
+    }
+
+    private fun handleGenericFile(reactContext: ReactContext?, intent: Intent) {
+        (intent.getParcelableExtra<Parcelable>(Intent.EXTRA_STREAM) as? Uri)?.let {
+            val params = Arguments.createMap()
+            params.putString("type", "file")
+            params.putString("sharedFileUri", it.toString())
+            if (reactContext == null) {
+                SharingModule.initialSharedData = params
+            } else {
+                sendEvent(reactContext, "shareReceived", params)
+            }
+        }
+    }
+
+    override fun onConfigurationChanged(newConfig: Configuration) {
+        super.onConfigurationChanged(newConfig)
+        val intent = Intent("onConfigurationChanged")
+        intent.putExtra("newConfig", newConfig)
+        this.sendBroadcast(intent)
+    }
+}
+
+
+fun launchMainActivity(context: Context) {
+    Log.d(com.zulipmobile.notifications.TAG, "Launching main activity")
+    val intent = Intent(context, MainActivity::class.java)
+    // See these sections in the Android docs:
+    //   https://developer.android.com/guide/components/activities/tasks-and-back-stack#TaskLaunchModes
+    //   https://developer.android.com/reference/android/content/Intent#FLAG_ACTIVITY_CLEAR_TOP
+    //
+    // * The flag FLAG_ACTIVITY_NEW_TASK is redundant in that it produces the
+    //   same effect as setting `android:launchMode="singleTask"` on the
+    //   activity, which we've done; but Context#startActivity requires it for
+    //   clarity's sake, a requirement overridden in Activity#startActivity,
+    //   because the behavior without it only makes sense when starting from
+    //   an Activity.  Our `context` is a service, so it's required.
+    //
+    // * The flag FLAG_ACTIVITY_CLEAR_TOP is mentioned as being what the
+    //   notification manager does; so use that.  It has no effect as long
+    //   as we only have one activity; but if we add more, it will destroy
+    //   all the activities on top of the target one.
+    intent.flags = Intent.FLAG_ACTIVITY_NEW_TASK or Intent.FLAG_ACTIVITY_CLEAR_TOP
+    context.startActivity(intent)
+}
```

Comment:
> The only reason I used it here was the `TAG` in `Log.d.` - I wanted the TAG to be the one declared in `ReceiveSharingActivity.ky`.

Hmm. For a bit when I was looking at this originally, I'd thought maybe that was the difference (though a small one). But what the code actually says is:
```kt
    Log.d(com.zulipmobile.notifications.TAG, "Launching main activity")
```
So it uses the same "TAG" value as in the other code. :slightly_smiling_face: 

(Which in itself I would have commented on if this function were otherwise good to merge, but that's a finer-grained matter.)